app/domain/authentication/authn_jwt/fetch_jwt_signing_key_factory.rb

```diff
@@ -0,0 +1,46 @@
+module Authentication
+  module AuthnJwt
+    # Factory that returns the interface implementation of FetchSigningKey
+    class FetchJwtSigningKeyFactory
+
+      def create(authenticator_parameters)
+        @authenticator_parameters = authenticator_parameters
+        validate_key_configuration
+
+        if provider_uri_has_valid_configuration?
+          Authentication::AuthnJwt::FetchProviderUriSigningKey.new(@authenticator_input,
```

Comment:
extract it to function `fetch_provider_uri_signing_key` or `fetch_provider_uri_signing_key_interface`